app/adapters/alma_adapter/availability_status.rb

```diff
@@ -12,61 +12,71 @@ def initialize(bib:)
 
     # Returns availability information for each of the holdings in the Bib record.
     def bib_availability
+      sequence = 0
       availability = holdings.each_with_object({}) do |holding, acc|
-        status = holding_status(holding)
-        acc[status[:id]] = status
+        sequence += 1
+        status = holding_status(holding: holding, sequence: sequence)
+        acc[status[:id]] = status if status.nil?
       end
       availability
     end
 
-    # TODO: handle these properties
-    #   "copy_number": item["copy_number"],
-    #   "patron_group_charged": nil,    # TODO: nil
-    #   "status": "???",                # TODO: "Not Charged"
-    # TODO: Confirm that we need all of these values with the Alma implementation.
-    # TODO: Find out if we can get copy_number from Solr (i.e. without hitting ExLibris' API)
-    def holding_status(holding)
+    # Returns availability information for each of the holdings in the Bib record.
+    # Notice that although we return the information by holding, we drill into item
+    # information to get details.
+    def bib_availability_from_items
+      availability = {}
+      item_data.each do |_key, value|
+        next if value.count == 0
+        raise StandardError.new, "Multiple items found under the same key" if value.count > 1
+        alma_item = AlmaAdapter::AlmaItem.new(Alma::BibItem.new(value.first.item))
+        status = holding_status_from_item(alma_item)
+        raise StandardError.new, "Holding found more than once" if availability[alma_item.holding_id]
+        availability[alma_item.holding_id] = status
+      end
+      availability
+    end
+
+    def holding_status(holding:, sequence:)
```

Comment:
This method is now greatly simplified because we are ONLY handling physical inventory and ignoring electronic and digital.